### PR TITLE
Various fixes to Search Page

### LINKED ISSUE #12: Implement the search page
Here are some items that still need to be done:
* [ ] Add hints and validation to the ASN input box
* [ ] If another search query is done we should terminate previous queries (see: #20)
* [x] Only list countries where we actually have measurements in 
* [x] Populate the list of test_names from a more canonical source (maybe the API?)

[Link to mockup](https://design.ooni.io/mockups/explorer/screen/e84dd189-cece-4495-9982-637ca77eed2d/Search-1/)

Currently the search page is done in part, however it's not fully aligned to the mockups.

In particular some of the form elements are not the same as the mockups. These components should eventually make their way into the `design-system`.

### PR DESCRIPTION
**Pull request by sarathms**
Fixes #12 

## Proposed Changes
(from #12)
* [x] Start date & end date filters don't work as expected (the filtering does not work and the address bar is not rewritten)
* [x] Date filter selectors should validate that the date range is consistent (i.e. start_date < end_date < NOW())
* [ ] Add hints and validation to the ASN input box
* [ ] If another search query is done we should terminate previous queries (see: https://github.com/ooni/explorer/issues/20)
* [ ] Only list countries where we actually have measurements in (@hellais will document how to do this)
* [ ] Populate the list of test_names from a more canonical source (maybe the API?) (@hellais will document how to do this)

### CODE REVIEW

**Comment by sarathms** [comment]
Added changes made during the session with @elioqoshi. Some previewed changes like container width enhancements have to come in through updates to `ooni-components`.

File: components/Layout.js [hunk 1]
```diff
@@ -26,8 +26,9 @@ injectGlobal`
     margin: 0;
     padding: 0;
     font-family: "Fira Sans";
+    font-size: 14px;
     height: 100%;
-    background-color: ${theme.colors.white};
+    background-color: #ffffff;
```
Comment (hellais) [inline components/Layout.js]: As mentioned via voice I think it's fine to not use constants for things like #fff (pure white) or #000 (pure black)

File: components/footer.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ import { injectIntl, intlShape } from 'react-intl'
 const StyledFooter = styled.footer`
   background-color: ${props => props.theme.colors.gray1};
   color: ${props => props.theme.colors.blue9};
-  font-size: 0.8rem;
+  font-size: 0.857rem;
```
Comment (hellais) [inline components/footer.js]: What is the origin of this odd font-size?

**Comment by hellais** [comment]
This looks good to me and I will merge. I will document the remaining issues as part of https://github.com/ooni/explorer/issues/12

File: components/footer.js (hunk 2 above)
Comment (sarathms) [reply components/footer.js]: When I first wrote the footer, the font size looked a bit big for smaller screens and so ended up using the `rem` notation to keep it relative. When @elioqoshi was reviewing it he found the font-size as `11.2px` where he expected `12px`. But he also agreed that using rem makes sense. Hence this value.

**Pull request merged**